textworld/generator/maker.py

```diff
@@ -682,7 +688,41 @@ def new_fact(self, name: str, *entities: List["WorldEntity"]) -> None:
         args = [entity.var for entity in entities]
         return Proposition(name, args)
 
-    def new_event_using_commands(self, commands: List[str]) -> Event:
+    def new_action(self, name: str, *entities: List["WorldEntity"]) -> Union[None, Action]:
+        """ Create new fact about a rule.
+
+        Args:
+            name: The name of the rule which can be used for the new rule fact as well.
+            *entities: A list of entities as arguments to the new rule fact.
+        """
+        if name not in self._kb.rules:
+            raise ValueError("Can't find action: '{}'".format(name))
+
+        rule = self._kb.rules[name]
+        mapping = {Placeholder(entity.type): Placeholder(entity.id, entity.type) for entity in entities}
+        return rule.substitute(mapping)
```

Comment:
@HakiRose This is how I find the corresponding "partial" Rule given a name and list of entities.